From liodot@gmail.com  Fri Oct 10 14:55:45 2008
From: "Lior Dotan" <liodot@gmail.com>
Date: Wed, 8 Oct 2008 11:37:37 +0200
Subject: Staging: SLICOSS: Call pci_release_regions at driver exit
To: "Greg KH" <greg@kroah.com>
Message-ID: <a5bf86a80810080237k1f197d8fv4f9ebed5b880df5c@mail.gmail.com>

slic_entry_probe() calls pci_request_regions() but there's no matching
pci_release_regions() at driver's exit or if slic_entry_probe() fails.

Signed-off-by: Lior Dotan <liodot@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/staging/slicoss/slicoss.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/staging/slicoss/slicoss.c
+++ b/drivers/staging/slicoss/slicoss.c
@@ -520,6 +520,7 @@ err_out_free_mmio_region:
 	release_mem_region(mmio_start, mmio_len);
 
 err_out_exit_slic_probe:
+	pci_release_regions(pcidev);
 	DBG_ERROR("%s EXIT jiffies[%lx] cpu %d\n", __func__, jiffies,
 		  smp_processor_id());
 
@@ -649,6 +650,7 @@ static void __devexit slic_entry_remove(
 	}
 	DBG_MSG("slicoss: %s deallocate device\n", __func__);
 	kfree(dev);
+	pci_release_regions(pcidev);
 	DBG_MSG("slicoss: %s EXIT\n", __func__);
 }
 
